app/components/NetworkRender.tsx

```diff
@@ -16,14 +16,14 @@ export default function NetworkRender({
 }) {
   function getContents() {
     if (filteredNetwork(network) !== chainName?.toLowerCase()) {
-      if (typeof switchNetwork === "function" && filteredNetwork(network) !== "hardhat") {
+      if (typeof switchNetwork === "function") {
         return (
           <div
             className={`tw-flex tw-flex-wrap tw-max-w-xs tw-text-sm tw-mx-auto ${children ? "tw-justify-center" : ""}`}
           >
             <p className="tw-mb-2"> You are currently connected to {chainName}</p>
             <button
-              onClick={() => switchNetwork()}
+              onClick={() => switchNetwork(chainId)}
```

Comment:
This is not entirely necessary as the chain id argument is set when we declare the switchNetwork hook in Wrapper.

Tested it locally and if we wanted to remove the chain Id prop being passed from Wrapper -> Rewards Header -> Network Render it would also work fine.